MoreLinq/StartsWith.cs

```diff
@@ -0,0 +1,49 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2012 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///     Determines whether the beginning of <paramref name="first" /> matches <paramref name="second" />.
+        /// </summary>
+        /// <typeparam name="T1">Type of elements in <paramref name="first" />.</typeparam>
+        /// <typeparam name="T2">Type of elements in <paramref name="second" />.</typeparam>
+        /// <param name="first">The sequence to check.</param>
+        /// <param name="second">The sequence to compare to.</param>
+        /// <returns>
+        ///     <c>true</c> if <paramref name="first" /> begins with elements equivalent to <paramref name="second" />.
+        /// </returns>
+        /// <remarks>
+        ///     This is the <see cref="IEnumerable{T}" /> equivalent of <see cref="string.StartsWith(string)" /> and
+        ///     it calls <see cref="object.Equals(object)" /> on pairs of elements at the same index.
+        /// </remarks>
+        public static bool StartsWith<T1, T2>(this IEnumerable<T1> first, IEnumerable<T2> second)
+        {
+            if (first == null) throw new ArgumentNullException("first");
+            if (second == null) throw new ArgumentNullException("second");
+
+            var sourceIter = first.GetEnumerator();
```

Comment:
This enumerator will leak as it is never disposed. I suggest adding a test to prove that and then wrapping it in `using`. In fact, the test should assert that both enumerators are properly disposed and while `All` is probably doing that for the second one, it's still good to have a test for in case someone refactors the implementation to not use `All`.